proposals/lip-0041.md

```diff
@@ -170,6 +167,96 @@ This function verifies the nonce of a given transaction `trs` and returns a bool
+  * The keys in the `authAccount.storeValue.mandatoryKeys` array are pairwise distinct and ordered lexicographically. The array may be empty.
+  * The keys in the `authAccount.storeValue.optionalKeys` array are pairwise distinct and ordered lexicographically. The array may be empty.
+  * The arrays `authAccount.storeValue.mandatoryKeys` and `authAccount.storeValue.optionalKeys` are disjoint; their union has at least one element and at most 64 elements.
+  * `authAccount.storeValue.numberOfSignatures` is an integer between 1 and the total number of public keys of the account (sum of the length of `authAccount.storeValue.mandatoryKeys` and `authAccount.storeValue.optionalKeys`).
+  * `authAccount.storeValue.numberOfSignatures` is not smaller than the length of `authAccount.storeValue.mandatoryKeys`.
+* For each entry `authAccount` in `genesisBlockAssetObject.authDataSubstore.storeEntries`, create a corresponding entry in the auth data substore with `storeKey = authAccount.storeKey` and `storeValue = authAccount.storeValue`.
+
+
+```java
+genesisAuthStoreSchema = {
+  "type": "object",
+  "required": ["authDataSubstore"],
+  "properties": {
+    "authDataSubstore": {
+      "type": "object",
+      "required": ["storePrefix", "storeEntries"],
+      "properties": {
+        "storePrefix": {
+          "dataType": "bytes",
+          "fieldNumber": 1
+        },
+        "storeEntries": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": ["storeKey", "storeValue"],
+            "properties": {
+              "storeKey": {
+                "dataType": "bytes",
+                "fieldNumber": 1
+              },
+              "storeValue": {
+                "type": "object",
+                "required": [
+                  "nonce",
+                  "numberOfSignatures",
+                  "mandatoryKeys",
+                  "optionalKeys"
+                ],
+                "properties": {
+                  "nonce": {
+                    "dataType": "uint64",
+                    "fieldNumber": 1
+                  },
+                  "numberOfSignatures": {
+                    "dataType": "uint32",
+                    "fieldNumber": 2
+                  },
+                  "mandatoryKeys": {
+                    "type": "array",
+                    "items": {
+                      "dataType": "bytes"
+                    },
+                    "fieldNumber": 3
+                  },
+                  "optionalKeys": {
+                    "type": "array",
+                    "items": {
+                      "dataType": "bytes"
+                    },
+                    "fieldNumber": 4
+                  }
+                },
+                "fieldNumber": 2
+              }
+            }
+          },
+          "fieldNumber": 2
```

Comment:
maybe fieldNumber can come as first prop so that it improves readability

Reply:
I would still keep it at the end as this is the template we chose to follow.

Reply:
nevermind, I was wrong. The `fieldNumber` should go second as you suggested, updated